app/src/main/java/tech/relaycorp/courier/domain/PublicSync.kt

```diff
@@ -29,14 +30,17 @@ class PublicSync
     }
 
     private suspend fun syncUnhandled() {
-        state.send(State.DeliveringCargo)
-        cargoDelivery.deliver()
+        val dohClient = DoHClient()
```

Comment:
I considered injecting this value, but decided against it for two reasons:

- It's only needed when synchronising with public gateways.
- I wanted to be able to release the underlying resources (e.g., the connection pool) once the public sync finished.

Reply:
You can always inject a Provider<DoHClient>. That way you can create and clean-up when you want, but still mock it in tests if needed.

Reply:
Ha, I totally forgot about providers! Been using TypeScript too much lately :)